aten/src/ATen/native/nested/NestedTensorMath.cpp

```diff
@@ -665,6 +665,80 @@ Tensor& NestedTensor_mul__Scalar(Tensor& self, const Scalar& other) {
   return NestedTensor_mul__Tensor(self, wrapped_scalar_tensor(other));
 }
 
+// Very rudimentary sum_dim for prototyping with torch_scatter.segment_reduce.
+Tensor NestedTensor_sum__dim(
+    const Tensor& self,
+    OptionalIntArrayRef opt_dims,
+    bool keepdim,
+    c10::optional<ScalarType> dtype) {
+  // Only allow reductions across the ragged (last) dim
```

Comment:
Note that NT supports raggedness in dims that are not the last dim

Reply:
Just to make sure I'm understanding this correctly, is it the case that only 1 dimension can be ragged but this dimension could be any dim?

Reply:
NT also supports multiple ragged dims for any dim but the first (i.e. the implicit batch dimension). so you could have:

```python
torch.nested_tensor([torch.randn(2, 3), torch.randn(4, 5)])
```

where both dims 1 and 2 are ragged.

Reply:
Ah ok, that was what I thought! I only support the last dim because in the above example I don't think we would be able to sum across dim=1 unless we treat the tensors as being implicitly padded with 0s, whereas we can always sum across dim=-1. Does that sound right?

My bad, I think my comment here was misleading.

Reply:
And yes the test cases should be testing this case since for example 

`self.random_nt(device, dtype, 4, (4, 4))` would generate

```
nested_tensor([
  tensor([], size=(0, 0)),
  tensor([], size=(0, 1)),
  tensor([[-0.0821],
          [ 0.6607],
          [ 0.0785]]),
  tensor([[ 0.7884,  0.1021],
          [-0.2590, -0.1549]])
])
```